frappe/desk/form/assign_to.py

```diff
@@ -197,4 +197,19 @@ def notify_assignment(assigned_by, owner, doc_type, doc_name, action='CLOSE',
 	enqueue_create_notification(owner, notification_doc)
 
 def format_message_for_assign_to(users):
-	return "<br><br>" + "<br>".join(users)
\ No newline at end of file
+	return "<br><br>" + "<br>".join(users)
+
+def update_todo_status(doctype, doc):
+	'''update todo doc status when reference's status changes'''
+	if hasattr(doctype, 'status') and doctype.status in ['Completed', 'Closed']:
+		todo_list = frappe.get_all('ToDo', ['name', 'status'], {'reference_name': doctype.name})
+
+		for todo in todo_list:
+			if todo.status == doctype.status:
+				return
+		
+			else:
+				frappe.db.set_value('ToDo', {'name': todo.name}, 'status', 'Closed')
```

Comment:
Seems a bit excessive to me, having to run an extra query on each Document update based on their status fields.